cleanlab/internal/util.py

```diff
@@ -688,6 +689,68 @@ def num_unique_classes(labels, multi_label=None) -> int:
         return len(set(labels))
 
 
+def format_labels(labels: LabelLike, multiannotator: bool = False) -> Tuple[LabelLike, dict]:
+    """Takes an array of labels and formats it such that labels are in the set ``0, 1, ..., K-1``,
+    where ``K`` is the number of classes. The labels are assigned based on lexicographic order.
+    Set ``multiannotator=True`` for datasets that have more than one annotator.
+
+    Returns
+    -------
+    labels
+        If ``multiannotator=False``, returns np.ndarray of shape ``(N,)``;
+        if ``multiannotator=True``, returns pd.DataFrame of shape ``(N,)``;
+        The return labels will be properly formatted and can be passed to other cleanlab functions.
+
+    mapping
+        A dictionary showing the mapping of new to old labels
+    """
+    if not multiannotator:
+        labels = labels_to_array(labels)
+        if labels.ndim != 1:
+            raise ValueError("labels must be 1D numpy array.")
+
+        unique_labels = np.unique(labels)
+        label_map = {label: i for i, label in enumerate(unique_labels)}
+        formatted_labels = np.array([label_map[l] for l in labels])
+        inverse_map = {i: label for label, i in label_map.items()}
+
+        return formatted_labels, inverse_map
+
+    else:
+        if isinstance(labels, pd.DataFrame):
+            np_labels = labels.values
+        elif isinstance(labels, np.ndarray):
+            np_labels = labels
+        else:
+            raise TypeError("labels must be 2D numpy array or pandas DataFrame")
+
+        unique_labels = pd.unique(np_labels.ravel())
+
+        try:
+            unique_labels = unique_labels[~np.isnan(unique_labels)]
+            unique_labels = np.sort(unique_labels)
+        except (TypeError):  # np.unique / np.sort cannot handle string values or pd.NA types
+            nan_mask = np.array(
+                [(l is np.NaN) or (l is pd.NA) or (l == "nan") for l in unique_labels]
+            )
+            unique_labels = unique_labels[~nan_mask]
+            unique_labels = np.sort(unique_labels)
```

Comment:
This can be done in-place with `np.ndarray.sort`([link](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.sort.html#numpy-ndarray-sort))

```suggestion
            unique_labels.sort()
        except (TypeError):  # np.unique / np.sort cannot handle string values or pd.NA types
            nan_mask = np.array(
                [(l is np.NaN) or (l is pd.NA) or (l == "nan") for l in unique_labels]
            )
            unique_labels = unique_labels[~nan_mask]
            unique_labels.sort()
```